Sources/StreamVideoSwiftUI/Livestreaming/LivestreamPlayer.swift

```diff
@@ -208,12 +323,29 @@ public struct LivestreamPlayer<Factory: ViewFactory>: View {
                 .foregroundColor(colors.livestreamCallControlsColor)
                 .overlay(
                     LivestreamDurationView(
-                        duration: viewModel.duration(from: state)
+                        duration: livestreamHelper.duration(from: state)
                     )
                 )
             }
         }
     }
+    
+    func joinLivestream() {
```

Comment:
what is happening if we move the app in the background and come back? Is that being triggered multiple times? Is it worth guarding against it happening on appear freely? (tbh the call internally shouldn't change state if it is already joined)

Reply:
makes sense, I will add some protection